File: src/Java/linked_list/List_test.java
package linked_list;

public class List_test {
    public static void main(String[] args) {
        ListNode node=new ListNode(1, new ListNode(2));
        ListNode test1=node, test2=node;
        test1.next=null;
        System.out.println((test2.next==null));
    }
}

File: src/C_Cpp/Linked_list/list_test.c
#include <stdio.h>
#include <stdlib.h>
#include "ListNode.h"

struct ListNode* MakeListNode(int val, struct ListNode* next){
    struct ListNode* node=(struct ListNode*)malloc(sizeof(struct ListNode));
    node->val=val;
    node->next=next;
    return node;
}

int main(){
    struct ListNode* tail=MakeListNode(2, NULL);
    struct ListNode* head=MakeListNode(1, NULL);
    head->next=tail;
    printf("%d -> %d\n", head->val, tail->val);
    return 0;
}